File: gear_generator/image_info.dart
import 'package:meta/meta.dart';

@immutable
class ImageInfo {
  ImageInfo(
      {this.outputWidth,
      this.outputHeight,
      this.svgInputFilePath,
      this.pngOutputFilePath});

  /// The width of the generated PNG file, in pixels
  final int outputWidth;

  /// The height of the generated PNG file, in pixels
  final int outputHeight;

  /// The absolute file path to the input SVG file
  final String svgInputFilePath;

  /// The absolut file path to the generated PNG file
  final String pngOutputFilePath;

  @override
  String toString() {
    return """ImageInfo(
  outputWidth: $outputWidth,
  outputHeight: $outputHeight,
  svgInputFilePath: $svgInputFilePath,
  pngOutputFilePath: $pngOutputFilePath
)""";
  }
}

File: gear_generator_2/src/image_info.ts
export interface ImageInfo {
  /**
   * The absolute path to the generated SVG file
   */
  svgPath: string;

  /**
   * The absolute path to the HTML file that hosts
   * the SVG file to be converted
   */
  htmlPath: string;

  /**
   * The absolute path to the rendered PNG file
   */
  pngPath: string;

  /**
   * The width of the image
   */
  width: number;

  /**
   * The height of the image
   */
  height: number;
}
